miniapps/nurbs/ex1.cpp

```diff
@@ -43,6 +43,95 @@
 using namespace std;
 using namespace mfem;
 
+
+/** Class for integrating the bilinear form a(u,v) := (Q Laplace u, v) where Q
+    can be a scalar coefficient. */
+class Diffusion2Integrator: public BilinearFormIntegrator
+{
+private:
+#ifndef MFEM_THREAD_SAFE
+   Vector shape,laplace;
+#endif
+   Coefficient *Q;
+
+public:
+   /// Construct a diffusion integrator with coefficient Q = 1
+   Diffusion2Integrator() { Q = NULL; }
+
+   /// Construct a diffusion integrator with a scalar coefficient q
+   Diffusion2Integrator (Coefficient &q) : Q(&q) { }
+
+   /** Given a particular Finite Element
+       computes the element stiffness matrix elmat. */
+   virtual void AssembleElementMatrix(const FiniteElement &el,
+                                      ElementTransformation &Trans,
+                                      DenseMatrix &elmat)
+   {
+      int nd = el.GetDof();
+      int dim = el.GetDim();
+      int spaceDim = Trans.GetSpaceDim();
+      bool square = (dim == spaceDim);
```

Comment:
This seems that it's not used anywhere

Reply:
Dim is used in one of the integration order computations, but the other 3 can indeed be deleted. 
I will delete when the style and compile issues are resolved.